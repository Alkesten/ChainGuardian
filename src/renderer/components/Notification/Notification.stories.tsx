import * as React from 'react';
import {useState} from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, number, select, button } from '@storybook/addon-knobs';
import { Notification} from './Notification';

storiesOf('Notification', module).add('System Notice/Error', () => {
    const [visible, setVisible]=useState(true);

    const levelOptions = {
        info: "info",
        error: "error",
    }
    const horizontalOptions = {
        left: "left",
        right: "right",
        center: "center-horizontal",
    }
    const verticalOptions = {
        top: "top",
        bottom: "bottom",
        center: "center-vertical",
    }
    // const visible = boolean("visible", true);

    const handler = () => setVisible(true);
    button("show notification", handler);

    const levelValue = select("level", levelOptions, "info");
    const horizontalValue = select("horizontalPosition", horizontalOptions, "left");
    const verticalValue = select("verticalPosition", verticalOptions, "top");
    const childrenValue = text("text", "Description of notice. Try to use conversational copy to give a friendly notice to your users!")

    return <Notification 
        title="Main message"
        isVisible={visible} 
        level={levelValue} 
        horizontalPosition={horizontalValue}
        verticalPosition={verticalValue}
        onClose={()=>setVisible(false)}>
        {childrenValue}
        </Notification>
}).addDecorator(withKnobs);