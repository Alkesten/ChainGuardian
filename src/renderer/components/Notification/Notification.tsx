import * as React from "react";

export interface INotificationProps{
    isVisible: boolean;
    level: level;
    horizontalPosition?: horizontal;
    verticalPosition?: vertical;
    title?: string;
    children?: any;
    onClose: () => void;
}

export enum level {
    info = "info",
    error = "error",
}

export enum horizontal {
    left = "left",
    right = "right",
    center = "center-horizontal",
}

export enum vertical {
    top = "top",
    bottom = "bottom",
    center = "center-vertical",
}

export const Notification: React.FunctionComponent<React.PropsWithChildren<INotificationProps>> = (
    props: INotificationProps) => {
    
    function getClasses(props: INotificationProps): string {
        return `${props.level} 
        ${props.isVisible ? "visible":"none"} 
        ${props.horizontalPosition} 
        ${props.verticalPosition}`;
    }

    return(
        <div className={`notification-container
        ${getClasses(props)} `}>
            <div className="notification-title">
                <div>{props.title}</div>
                <div onClick={(): void => props.onClose()} className="close-icon" />
            </div>
            <div className="notification-content">
                {props.children}
            </div>
        </div>
    );
};