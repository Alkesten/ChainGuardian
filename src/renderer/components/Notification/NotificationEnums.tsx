export enum level {
    INFO = "info",
    ERROR = "error",
}

export enum horizontal {
    LEFT = "left",
    RIGHT = "right",
    CENTER = "center-horizontal",
}

export enum vertical {
    TOP = "top",
    BOTTOM = "bottom",
    CENTER = "center-vertical",
}