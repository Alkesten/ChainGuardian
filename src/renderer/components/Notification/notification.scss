.notification-stacked-container{
    position: absolute;
    width: 640px;
    height: 100%;
    display: flex;
    flex-direction: column-reverse;
    pointer-events: none;
    &.right{
        right: 32px;
    }
    &.bottom{
        bottom: 32px;
    }
}
.notification-stacked-container .notification-stacked{
    pointer-events: auto;
}
.notification-stacked{
    border-radius: 8px;
    width: 576px;
    padding: 32px;
    margin-top: 10px;
    font-family: 'Palanquin', sans-serif;
    z-index: 1000;
    //level style-----------------------
    &.info{
        border-top: 4px solid $secondary-blue-400;
    }
    &.error{
        border-top: 4px solid $system-error-500;
    }
    &.info .notification-title{
        color: $secondary-blue-100;
    }
    &.error .notification-title{
        color: $system-error-100;
    }
    &.info .notification-content{
        color: $secondary-blue-200;
    }
    &.error .notification-content{
        color: $system-error-200;
    }
    // status
    &.visible{
        display: block;
    }
    &.none{
        display: none;
    }
}

.notification{
    position: absolute;
    border-radius: 8px;
    width: 576px;
    padding: 32px;
    font-family: 'Palanquin', sans-serif;

    &.visible{
        display: block;
    }
    &.none{
        display: none;
    }

    //position horizontal-----------------------
    &.left{
        left: 32px;
    }
    &.right{
        right: 32px;
    }
    &.center-horizontal{
        margin-left: auto;
        margin-right: auto;
        left: 0;
        right: 0;
    }

    //positon vertical-----------------------
    &.top{
        top: 32px;
    }
    &.bottom{
        bottom: 32px;
    }
    &.center-vertical{
        top: calc(50vh - 100px);
    }

    //level style-----------------------
    &.info{
        border-top: 4px solid $secondary-blue-400;
    }
    &.error{
        border-top: 4px solid $system-error-500;
    }
    &.info .notification-title{
        color: $secondary-blue-100;
    }
    &.error .notification-title{
        color: $system-error-100;
    }
    &.info .notification-content{
        color: $secondary-blue-200;
    }
    &.error .notification-content{
        color: $system-error-200;
    }

}
.notification-title{
    font-weight: 500;
    font-size: 24px;
    line-height: 28px;
    letter-spacing: 0.01em;
    display: flex;
    justify-content: space-between;
}
.close-icon{
    background: url(../assets/img/notification/Close.svg);
    height: 24px;
    width: 24px;
}
.notification-content{
    font-weight: 400;
    font-size: 18px;
    line-height: 28px;
}
.notification-level-info{
    background-color: $secondary-blue-600;
}
.notification-level-error{
    background-color: $system-error-700;
}
