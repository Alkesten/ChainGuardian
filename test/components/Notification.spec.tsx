import * as React from "react";
import * as renderer from "react-test-renderer";
import {Notification} from "../../src/renderer/components/Notification/Notification";
import {level, horizontal, vertical} from "../../src/renderer/components/Notification/NotificationEnums";

describe("Notification", () => {
    it("renders info top left", () => {
        const tree = renderer
            .create(<Notification 
                title="Test Title"
                isVisible={true}
                level={level.INFO}
                horizontalPosition={horizontal.LEFT}
                verticalPosition={vertical.TOP}
                onClose={(): void=>{}}>
                Test text for notification commponent
            </Notification>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders error bottom right", () => {
        const tree = renderer
            .create(<Notification 
                title="Test Title"
                isVisible={true}
                level={level.ERROR}
                horizontalPosition={horizontal.RIGHT}
                verticalPosition={vertical.BOTTOM}
                onClose={(): void=>{}}>
                Test text for notification commponent
            </Notification>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders hidden notification", () => {
        const tree = renderer
            .create(<Notification 
                title="Test Title"
                isVisible={false}
                level={level.INFO}
                horizontalPosition={horizontal.RIGHT}
                verticalPosition={vertical.BOTTOM}
                onClose={(): void=>{}}>
                Test text for notification commponent
            </Notification>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders centered notification", () => {
        const tree = renderer
            .create(<Notification 
                title="Test Title"
                isVisible={true}
                level={level.INFO}
                horizontalPosition={horizontal.CENTER}
                verticalPosition={vertical.CENTER}
                onClose={(): void=>{}}>
                Test text for notification commponent
            </Notification>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});