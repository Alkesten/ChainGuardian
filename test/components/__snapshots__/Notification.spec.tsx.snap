// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders centered notification 1`] = `
<div
  className="notification-container
        visible center-horizontal center-vertical info "
>
  <div
    className="notification-title"
  >
    <div>
      Test Title
    </div>
    <div
      className="close-icon"
      onClick={[Function]}
    />
  </div>
  <div
    className="notification-content"
  >
    Test text for notification commponent
  </div>
</div>
`;

exports[`Notification renders error bottom right 1`] = `
<div
  className="notification-container
        visible right bottom error "
>
  <div
    className="notification-title"
  >
    <div>
      Test Title
    </div>
    <div
      className="close-icon"
      onClick={[Function]}
    />
  </div>
  <div
    className="notification-content"
  >
    Test text for notification commponent
  </div>
</div>
`;

exports[`Notification renders hidden notification 1`] = `
<div
  className="notification-container
        none right bottom info "
>
  <div
    className="notification-title"
  >
    <div>
      Test Title
    </div>
    <div
      className="close-icon"
      onClick={[Function]}
    />
  </div>
  <div
    className="notification-content"
  >
    Test text for notification commponent
  </div>
</div>
`;

exports[`Notification renders info top left 1`] = `
<div
  className="notification-container
        visible left top info "
>
  <div
    className="notification-title"
  >
    <div>
      Test Title
    </div>
    <div
      className="close-icon"
      onClick={[Function]}
    />
  </div>
  <div
    className="notification-content"
  >
    Test text for notification commponent
  </div>
</div>
`;
